### ISSUE TITLE
AzerothCore does not support MariaDB below 10.5

### ISSUE THREAD

**Issue by virgo77** [azerothcore/azerothcore-wotlk#9831]
### Current Behaviour

Hi all! Me again :)

I'm trying to migrate the database from MySQL8 to MariaDB 10.5.12 and even AuthServer or WorldServer won't connect to the SQL Server.

My Homelab:
- AzerothCore: Windows Server 2022 Std / 6 vCPU / 24 GB RAM / MySQL8
- SQL Server: Debian 11 / MariaDB 10.5.12 updated / Local DB User (remote connection OK)

I don't know if the error causes the SQL no connection. I tried to update MariaDB 10.5 to MariaDB 10.6 and the upgrade fails. I followed the official tutorial here: https://mariadb.com/docs/operations/upgrades/upgrade-community-server-cs10-6-debian11/ and I got a curl error "curl: (23) Failure writing output to destination"
I tried to search why CURL is in error...

Here is a PasteBin log : https://pastebin.com/5f7xrS5Q

### Expected Blizzlike Behaviour

having no errors
Being able to connect AC to the DB

### Steps to reproduce the problem

Just launch the latest AC Updated and compiled
It's a behavior I got since one week yet

### AC rev. hash/commit

AzerothCore rev. c70cc855ff3d 2021-12-19 22:23:55 -0300 (master branch) (Win64, RelWithDebInfo, Static)

### Operating system

Windows Server 2022 Standard, Debian 11, MariaDB 10.5.12

### Custom changes or Modules

no real mod, but just on the worldserver.conf file:
- PreloadAllNonInstancedMapGrids = 1
- SetAllCreaturesWithWaypointMovementActive = 1

**Comment by yuanf225** [comment]
use MariaDB 10.6

**Comment by Winfidonarleyan** [comment]
Please use same libs for client and server

**Comment by virgo77** [comment]
> use MariaDB 10.6

Ah Mr Obvious. It's clear that you did not read my post... many help would be appreciated instead of this one. So next time, please keep quiet. Thanks a lot



> Please use same libs for client and server

Hi Winfidonarleyan (not easy to write, huh ^^)
What do you mean exactly? Thanks :)

**Comment by Kitzunu** [comment]
First of all. Use a better tone really.
Second of all, several users use MariaDB 10.5 and 10.6 with no issues. 

So how do you troubleshoot this.
Try reinstalling your MariaDB Server and Client. And make sure you use the EXACT .dll files provided by the installs in your server folder or it will fail.

**Comment by virgo77** [comment]
Hi Kitzunu
Thanks for your reply. Don't worry: I'm really fine. The tone I used is just against Yuanf225 that was not really helping me with his answer which is... obvious! And as I said in my post, upgrading to MariaDB 10.6 fails because of a curl error I can't resolve. And, his answer was not really the one I would expect. So, I apologize if my words offended you.


About your help: I don't really understand what you mean about the "DLLs provided by the install". So it's a bit hard for me to try to troubleshoot and tell you what I've done for.

_[... ha damn... as I was writing a very long explanation to tell you how I used the AC wikis ans how I made the compile, etc. to make my own WoW Server, I think I understand what you want to tell me...]_

Don't you ask about the 'libeay32.dll', 'libmysql.dll', 'ssleay32.dll' that are on the /server/config folder? If so, is there a way to install the MariaDB MSI File whereas MySQL8 is still installed? And, is the 10.6.13-GA x64 (the core is compiled in x64) can be used whereas my Debian MariaDB is in 10.5.12?
I'm just thinking about CMAKE and VS2019: I'm not sure that the SQL DLLs are required during compilation, because in CMAKE, the checkbox 'USE_MYSQL_SOURCES' is unticked by default. An advice?

Thanks for your reply. Maybe it's not as hard as I'm thinking it is.
And sorry for my bad english if I'm not always understandable :)

**Comment by Eronox** [comment]
Your output implies it's still trying to use an MySQL instance.

> MySQL client library: 8.0.26
MySQL server ver: 5.5.5-10.5.12-MariaDB-0+deb11u1-log 
Connected to MySQL database at 192.168.1.231

In the build folder, inside the config folder, replace `libmysql.dll` with the `libmysql.dll` you find in your MariaDB installation folder.

**Comment by virgo77** [comment]
Hi Eronox,
Thanks for your help.
Erm, in the /core/build/ folder, I don't have any 'configs' folder or DLL file, sorry.

So, should I put these ones
   libmariadb.dll
   libmariadb.lib
   mariadbclient.lib
   mysqlservices.lib
   libmariadb.pdb
and then recompile? Or just copy these files to the /server/configs folder? I tried this option, and it fails too

Or maybe I need to tick the 'USE_MYSQL_SOURCES' in CMAKE?

Thanks a lot

**Comment by yuanf225** [comment]
Haha, your tone is not important to me. If you read the wiki carefully, you will know that when you update the database and database environment, it is best to recompile with the new environment. In addition, libmariadb.dll files are installed in mariadb  In the directory

**Comment by Eronox** [comment]
> Hi Eronox, Thanks for your help. Erm, in the /core/build/ folder, I don't have any 'configs' folder or DLL file, sorry.

The DLL's are not there during compilation, but instead for running worldserver.exe.

For example, my source is at `azerothcore/src` and my build is at `azerothcore/build`.
And my DLL's are here: `zerothcore\build\bin\Debug`.  Debug may be release for you, depending on what you compile into.

**Comment by acidmanifesto** [comment]
As this issue can not be replicated, i will be closing the issue. Feel Free to open up another issue if the matter comes up again. But as far as we can tell its a library issue on the user end not matching that with the database as mentioned earlier. Be sure to follow the wiki to ensure all steps are followed verbatim.
https://www.azerothcore.org/wiki/

**Comment by virgo77** [comment]
Hi All!
First of all: Merry Christmas
Secondly: I would like to thank those who answered me and who, without giving me the solution directly, forced me to think a little and understand what I had to do.
And I inform that I succeeded! it works fine and I no longer have an incompatibility error between Auth / World and MariaDB 10.5. No need to upgrade to MariaDB 10.6. So thank you!
Thirdly: I am looking for a solution because I have another small error which does not seem annoying to me, but which I would like to correct, if a solution exists. Maybe I will open a new topic if I don't find any solution.

Lastly and once again: thank you and merry christmas! 

V.
